packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -1683,18 +1683,46 @@ class RoundedRectSliderTrackShape extends SliderTrackShape with BaseSliderTrackS
     );
 
     // The arc rects create a semi-circle with radius equal to track height.
-    final Rect leftTrackArcRect = Rect.fromLTWH(trackRect.left, trackRect.top, trackRect.height, trackRect.height);
+    final Rect leftTrackArcRect = Rect.fromLTWH(
+      trackRect.left - trackRect.height / 2,
+      trackRect.top,
+      trackRect.height,
+      trackRect.height,
+    );
```

Comment:
moved the ARC to the left by height/2, as the thumb should not go to the edge as it was before